plugins/woocommerce/src/Blocks/Utils/BlockTemplateUtils.php

```diff
@@ -690,6 +690,45 @@ function( $template ) use ( $customised_template_slugs ) {
 		);
 	}
 
+	/**
+	 * Removes customized templates that shouldn't be available. That means customized templates from other themes or
+	 * customized templates based on the WooCommerce default templates when there is a customized template based on the
+	 * theme one.
+	 *
+	 * @param \WP_Block_Template[]|\stdClass[] $templates  List of templates to run the filter on.
+	 * @param string                           $theme_slug Slug of the theme currently active.
+	 *
+	 * @return array Filtered list of templates with only relevant templates available.
+	 */
+	public static function remove_duplicate_customized_templates( $templates, $theme_slug ) {
+		$filtered_templates = array_filter(
+			$templates,
+			function( $template ) use ( $templates, $theme_slug ) {
+				if ( $template->theme === $theme_slug ) {
+					// This is a customized template based on the theme template, so it should be returned.
+					return true;
+				} elseif ( self::PLUGIN_SLUG !== $template->theme && self::DEPRECATED_PLUGIN_SLUG !== $template->theme ) {
```

Comment:
Do we need this? In `get_block_templates_from_db`, we only get the templates with theme set to Woo and the current theme, so I'm afraid this will be never executed 🤔